x-pack/plugins/observability/public/pages/rules/index.tsx

```diff
@@ -17,20 +17,26 @@ import { Provider, rulesPageStateContainer, useRulesPageStateContainer } from '.
 
 import { useBreadcrumbs } from '../../hooks/use_breadcrumbs';
 import { useKibana } from '../../utils/kibana_react';
-import { OBSERVABILITY_SOLUTIONS } from './config';
 import { RULES_PAGE_TITLE, RULES_BREADCRUMB_TEXT } from './translations';
 
+const RULES_LIST_COLUMNS_KEY = 'observability_rulesListColumns';
+
 function RulesPage() {
-  const { ObservabilityPageTemplate } = usePluginContext();
+  const { ObservabilityPageTemplate, observabilityRuleTypeRegistry } = usePluginContext();
   const { http, docLinks, triggersActionsUi } = useKibana().services;
 
   const documentationLink = docLinks.links.observability.createAlerts;
 
+  const filteredRuleTypes = useMemo(
+    () => observabilityRuleTypeRegistry.list(),
+    [observabilityRuleTypeRegistry]
+  );
+
   const { status, setStatus, lastResponse, setLastResponse } = useRulesPageStateContainer();
   const [createRuleFlyoutVisibility, setCreateRuleFlyoutVisibility] = useState(false);
   const [refresh, setRefresh] = useState(new Date());
   const { ruleTypes } = useLoadRuleTypes({
-    filteredSolutions: OBSERVABILITY_SOLUTIONS,
```

Comment:
@XavierM `useLoadRuleTypes` expected an array of observability solutions for example `['logs', 'apm', ...]`, whereas `observabilityRuleTypeRegistry.list()` returns an array of the observability registered rule types (id of the rule type when registering it to the alerting framework), for example `['xpack.uptime.alerts.monitorStatus', 'xpack.uptime.alerts.tls', 'apm.error_rate', 'metrics.alert.inventory.threshold', ...]`, so I am wondering how this change could even work.